internal/lsp/diagnostics.go

```diff
@@ -17,3 +17,24 @@ func HCLSeverityToLSP(severity hcl.DiagnosticSeverity) lsp.DiagnosticSeverity {
 	}
 	return sev
 }
+
+func HCLDiagsToLSP(hclDiags hcl.Diagnostics) []lsp.Diagnostic {
+	diags := []lsp.Diagnostic{}
+
+	for _, hclDiag := range hclDiags {
+		// only process diagnostics with an attributable spot in the code
+		if hclDiag.Subject != nil {
+			msg := hclDiag.Summary
+			if hclDiag.Detail != "" {
+				msg += ": " + hclDiag.Detail
+			}
+			diags = append(diags, lsp.Diagnostic{
+				Range:    HCLRangeToLSP(*hclDiag.Subject),
+				Severity: HCLSeverityToLSP(hclDiag.Severity),
+				Source:   "Terraform",
```

Comment:
Previously I had this set to `HCL`, however now I think since the diagnostics are coming from the decoder, they could be interpreted as not just syntax, but also provider schema specific? Therefore I thought saying the source was `Terraform` was most appropriate, let me know if that is an incorrect interpretation

Reply:
The [spec](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#diagnostic) doesn't seem to go into much detail in describing this field, but it seems to suggest that each linter should be reported with its own `Source` - in our case I reckon that would apply to `tflint` and I would think potentially to `terraform validate` too?

The key question is how does the field affect the UX or anything in practice - I have no idea 🤷‍♂️ 
So let's keep `Terraform` there for now and we can revisit that decision later as we expand diagnostics support.

Reply:
I'll switch it back then.